enacts/flex_fcst/cpt.py

```diff
@@ -101,62 +416,153 @@ def starts_list(
         start_date = re.search(regex_search_pattern, file)
         start_date_dt = datetime.strptime(start_date.group(), format_in)
         start_dates.append(start_date_dt)
-    start_dates = sorted(set(start_dates)) #finds unique dates in the case there are files with the same date due to multiple lead times
+    #finds unique dates in the case there are files with the same date due to
+    # multiple lead times
+    start_dates = sorted(set(start_dates))
     start_dates = [i.strftime(format_out) for i in start_dates]
     return start_dates
 
 
 def read_pycptv2dataset(data_path):
+    """ Create a xr.Dataset from yearly pyCPT nc files for one or more targets of the
+    year, for a forecast mean and variance variables, appending multiple Starts; and
+    the corresponding xr.DataArray of the observations time series used to train
+    those targets
+
+    Parameters
+    ----------
+    data_path: str
+        path of set of nc files, organized by targets if multiple ones.
+
+    Returns
+    -------
+    fcst_ds, obs : xr.Dataset, xr.DataArray
+        dataset of forecast mean and variance from `data_path` and their obs
+
+    See Also
+    --------
+    read_pycptv2dataset_single_target
+
+    Notes
+    -----
+    Whether `data_path` contains multiple targets or not is determined by whether the
+    observation obs.nc file(s) is a direct child of `data_path` or under target-wise
+    folders that have no restriction on how they are named.
+    If multiple targets, `fcst_ds` is expanded in to a square with the L dimension,
+    and so are the Ts coordinates that are turned into variables.
+    If single target, `fcst_ds` remains 1D against S only and Ts remain coordiantes
+    of S.
+    `obs` is in any case always appended and sorted into a natural 1D time series
+    """
     data_path = Path(data_path)
     children = list(data_path.iterdir())
-
     if 'obs.nc' in map(lambda x: str(x.name), children):
-        fcst_mu, fcst_var, obs = read_pycptv2dataset_single_target(data_path)
+        fcst_ds, obs = read_pycptv2dataset_single_target(data_path)
```

Comment:
I had to change the rationale throughout from outputting tuples of xr.DataArrays to xr.Datasets of xr.DataArrays because:

1. so the outputs in the both nc/tsv cases be the same (tsv can also outputs hcst)
2. because I need to make ds-wise manipulations in the nc case where the Ts can either remain coordinates (single target) or become variables (multiple targets)